camel/agents/chat_agent.py

```diff
@@ -113,9 +142,17 @@ def __init__(
 
         self.model: ModelType = (model if model is not None else
                                  ModelType.GPT_3_5_TURBO)
-        self.model_config: ChatGPTConfig = model_config or ChatGPTConfig()
         self.message_window_size: Optional[int] = message_window_size
 
+        # This is a very dirty implementation. We should consider to
```

Comment:
emm this was written by Tianqi previously as he thought it might be possible to move the function calling part separate. But after our discussion we realized it is quite hard to decouple this part from `ChatAgent` running logic (as mentioned in the reasons why we need to include a while-loop in method `step). The decoupling was later partly solved by packing the logic of function calling into a separate function `step_function_call` but the loop-based logic seems necessary to be kept here.